### PR TITLE
ITE: soc/riscv: it8xxx2: Add it82xx2 related changes

### PR DESCRIPTION
**Pull request by GTLin08**
This PR adds it82xx2 chip related changes, including driver, chip configuration, devicetree, it82xx2_evb board created and unit test.

### CODE REVIEW

File: boards/riscv/it82xx2_evb/board.cmake [hunk 1]
```diff
@@ -0,0 +1,3 @@
+set(SUPPORTED_EMU_PLATFORMS renode)
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/board.cmake:1]: Which resc/repl files is this board using? Because this file is missing the `RENODE_SCRIPT` variable.

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,246 @@
+.. _it82xx2_evb:
+
+ITE IT82XX2 series
+######################
+
+Overview
+********
+
+The IT82XX2 is a 32-bit RISC-V Micro-controller.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:9]:
```suggestion
The IT82XX2 is a 32-bit RISC-V microcontroller.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+     - timer
+   * - WATCHDOG
+     - on-chip
+     - watchdog
+   * - KSCAN
+     - on-chip
+     - kscan
+   * - SENSOR
+     - on-chip
+     - voltate comparator
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:90]:
```suggestion
     - voltage comparator
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+In order to upload the application to the device,
+you'll need our flash tool and Download board.
+You can get them at: `ITE's website`_.
+
+Wiring
+=======
+#. Connect the Download Board to your host computer using the USB cable.
+
+#. Connect the it82xx2_evb to evolution motherboard.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:112]:
```suggestion
#. Connect the it82xx2_evb to the evolution motherboard.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+#. Connect the Download Board J5 to J41 on evolution motherboard.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:114]:
```suggestion
#. Connect the Download Board J5 to J41 on the evolution motherboard.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+
+#. Connect the USB to UART wire to J33 on evolution motherboard.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:116]:
```suggestion
#. Connect the USB to UART wire to J33 on the evolution motherboard.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+Flashing
+========
+
+Windows
+--------
+
+Use the winflash tool to program a zephyr application
+to the it82xx2 board flash.
+
+#. Open winflash tool and make sure the order you open the switch is right.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:173]:
```suggestion
#. Open the winflash tool and make sure the order you open the switch is right.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+   Fisrt, turn on the Download board switch.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:174]:
```suggestion
   First, turn on the Download board switch.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+   Second, turn on the it82xx2_evb board switch.
+   Then, configure your winflash tool like below.
+
+    .. figure:: WinFlashTool_P2.jpg
+          :align: center
+
+    .. figure:: WinFlashTool_P4.jpg
+
+#. Using winflash tool flash zephyr.bin into your ITE board.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:183]:
```suggestion
#. Using the winflash tool flash ``zephyr.bin`` into your ITE board.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+   First, click ``Load`` button and select your zephyr.bin file.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:184]:
```suggestion
   First, click the ``Load`` button and select your ``zephyr.bin`` file.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+   Second, click ``run`` to flash the iamge into board.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:185]:
```suggestion
   Second, click ``run`` to flash the image into board.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+
+    .. figure:: WinFlashTool_P3.jpg
+          :align: center
+
+#. At this point, you have flashed your image into ITE board and
+   it will work if you turn on ITE board. You can use a terminal program
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:191]:
```suggestion
   it will work if you turn on the ITE board. You can use a terminal program
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+#. Run your favorite terminal program to listen for output.
+   Under Linux the terminal should be :code:`/dev/ttyUSB0`. Do not close it.
+
+   For example:
+
+   .. code-block:: console
+
+      $ minicom -D /dev/ttyUSB0 -b 115200
+
+#. Open a second terminal window and use linux flash tool to flash your board.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:216]:
```suggestion
#. Open a second terminal window and use the Linux flash tool to flash your board.
```

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+    https://www.ite.com.tw/uploads/product_download/itedlb4-linux-v106.tar.bz2
+
+#. Split first and second terminal windows to view both of them.
+   You should see ``"Hello World! it82xx2_evb"`` in the first terminal window.
+   If you don't see this message, press the Reset button and the message should appear.
+
+Debugging
+=========
+
+Supporting uart debug, currently.
```
Comment (fkokosinski) [inline boards/riscv/it82xx2_evb/doc/index.rst:232]: Could you elaborate on that?

File: drivers/adc/adc_ite_it8xxx2.c [hunk 15]
```diff
@@ -30,7 +30,7 @@ LOG_MODULE_REGISTER(adc_ite_it8xxx2);
 /* ADC channels disabled */
 #define IT8XXX2_ADC_CHANNEL_DISABLED 0x1F
 /* ADC sample time delay (Unit:us) */
-#define IT8XXX2_ADC_SAMPLE_TIME_US 200
+#define IT8XXX2_ADC_SAMPLE_TIME_US 500
```
Comment (fkokosinski) [inline drivers/adc/adc_ite_it8xxx2.c:33]: Why is this change required? Is this sample time delay also valid on the `it8xxx2` board?

File: drivers/gpio/gpio_ite_it8xxx2_v2.c [hunk 16]
```diff
@@ -0,0 +1,460 @@
...
+		}
+	}
+
+	/* Set input or output. */
+	if (flags & GPIO_OUTPUT)
+		*reg_gpcr = (*reg_gpcr | GPCR_PORT_PIN_MODE_OUTPUT) &
+				~GPCR_PORT_PIN_MODE_INPUT;
+	else
+		*reg_gpcr = (*reg_gpcr | GPCR_PORT_PIN_MODE_INPUT) &
+				~GPCR_PORT_PIN_MODE_OUTPUT;
```
Comment (fkokosinski) [inline drivers/gpio/gpio_ite_it8xxx2_v2.c:162]: Please add braces even for single-line if bodies.

File: drivers/gpio/gpio_ite_it8xxx2_v2.c [hunk 17]
```diff
@@ -0,0 +1,460 @@
...
+	struct gpio_ite_data *data = dev->data;
+	uint8_t mask = BIT(pin);
+	gpio_flags_t flags = 0;
+
+	__ASSERT(gpio_config->index < GPIO_GROUP_COUNT + KBS_GROUP_COUNT,
+		 "Invalid GPIO group index");
+
+	/* push-pull or open-drain */
+	if (*reg_gpotr & mask)
+		flags |= GPIO_OPEN_DRAIN;
```
Comment (fkokosinski) [inline drivers/gpio/gpio_ite_it8xxx2_v2.c:199]: Ditto

**Review by fkokosinski — CHANGES_REQUESTED** [review]

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 2 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 3 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 4 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 5 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 6 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 7 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 8 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 9 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 10 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 11 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 12 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 13 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 14 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Done.

File: drivers/gpio/gpio_ite_it8xxx2_v2.c (hunk 16 above)
Comment (GTLin08) [reply drivers/gpio/gpio_ite_it8xxx2_v2.c:162]: Done.

File: drivers/gpio/gpio_ite_it8xxx2_v2.c (hunk 17 above)
Comment (GTLin08) [reply drivers/gpio/gpio_ite_it8xxx2_v2.c:200]: Done.

File: drivers/adc/adc_ite_it8xxx2.c (hunk 15 above)
Comment (GTLin08) [reply drivers/adc/adc_ite_it8xxx2.c:33]: There is an explanation in line 436 for the change in sampling time, which also applies to it8xxx2_evb board.

File: boards/riscv/it82xx2_evb/board.cmake (hunk 1 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/board.cmake]: West flash and west debug are not currently supported, so we can remove this file.

File: drivers/adc/adc_ite_it8xxx2.c (hunk 15 above)
Comment (fkokosinski) [reply drivers/adc/adc_ite_it8xxx2.c:33]: Got it, thanks

**Review by fkokosinski — DISMISSED** [review]
LGMT

File: soc/riscv/riscv-ite/common/check_regs.c [hunk 18]
```diff
@@ -147,6 +147,39 @@ IT8XXX2_REG_OFFSET_CHECK(adc_it8xxx2_regs, adc_vchs_ctrl[0].VCHCTL, 0x60);
...
+/* Watchdog register structure check */
+#ifdef CONFIG_SOC_IT8XXX2_REG_SET_V1
+IT8XXX2_REG_SIZE_CHECK(wdt_it8xxx2_regs, 0x62);
+IT8XXX2_REG_OFFSET_CHECK(wdt_it8xxx2_regs, ET1CNTOLR, 0x40);
+IT8XXX2_REG_OFFSET_CHECK(wdt_it8xxx2_regs, ET1CNTOHR, 0x41);
+IT8XXX2_REG_OFFSET_CHECK(wdt_it8xxx2_regs, ET2CNTOLR, 0x44);
+IT8XXX2_REG_OFFSET_CHECK(wdt_it8xxx2_regs, ET2CNTOHR, 0x45);
+IT8XXX2_REG_OFFSET_CHECK(wdt_it8xxx2_regs, ET2CNTOH2R, 0x46);
+IT8XXX2_REG_OFFSET_CHECK(wdt_it8xxx2_regs, EWDCNTOLR, 0x60);
+IT8XXX2_REG_OFFSET_CHECK(wdt_it8xxx2_regs, EWDCNTOHR, 0x61);
```
Comment (keith-zephyr) [inline soc/riscv/riscv-ite/common/check_regs.c:170]: It seems like the register space could be managed from the devicetree.

You should consider using multiple `reg` pairs to specify the base addresses for ETWCFG and ET1CNTOLR.

**Review by keith-zephyr — CHANGES_REQUESTED** [review]
I've reviewed up through the watchdog timer.  I will review the other pieces soon.

File: drivers/interrupt_controller/Kconfig.it8xxx2 [hunk 19]
```diff
@@ -10,6 +10,15 @@ config ITE_IT8XXX2_INTC
...
 	 this software interrupt default set on by device tree.
 
+config ITE_IT8XXX2_INTC_V2
+	def_bool DT_HAS_ITE_IT8XXX2_INTC_V2_ENABLED
+	depends on DT_HAS_ITE_IT8XXX2_INTC_V2_ENABLED
+	help
+	 Configures the maximum number of clients allowed per shared
+	 instance of the shared interrupt driver. To conserve RAM set
+	 this value to the lowest practical value.
+	 this software interrupt default set on by device tree.
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply drivers/interrupt_controller/Kconfig.it8xxx2]: Help text fixed. Yes, it enables V2 interrupt controller only.

File: drivers/interrupt_controller/Kconfig.it8xxx2 [hunk 20]
```diff
@@ -10,6 +10,15 @@ config ITE_IT8XXX2_INTC
 	 this value to the lowest practical value.
 	 this software interrupt default set on by device tree.
 
+config ITE_IT8XXX2_INTC_V2
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply drivers/interrupt_controller/Kconfig.it8xxx2:20]: Done.

File: soc/riscv/riscv-ite/common/check_regs.c (hunk 18 above)
Comment (GTLin08) [reply soc/riscv/riscv-ite/common/check_regs.c]: Remove these unused watchdog registers. If these are used later, we will add them according to your method.

**Comment by GTLin08** [comment]
In addition, this push has removed the redundant declaration as below:
```
config ITE_IT8XXX2_INTC
	default y
```
in Kconfig.defconfig.it81202bx, Kconfig.defconfig.it81202cx, Kconfig.defconfig.it81302bx, Kconfig.defconfig.it81302cx,
    Kconfig.defconfig.it82202ax, Kconfig.defconfig.it82302ax,

Path as below: [compare](https://github.com/zephyrproject-rtos/zephyr/compare/e02d16d37d88a69610c51b69d510a2f5af1111ab..cbe2a1b7428e78d38cb5a18ff6ddc3f4916098a1)

File: boards/riscv/it82xx2_evb/doc/index.rst [hunk 21]
```diff
@@ -0,0 +1,247 @@
...
+
+Hardware
+********
+The IT82XX2 series contains different chip types(ex, it82202, it82302),
+and they support different hardware features.
+Listing the IT82202 hardware features as following:
+
+- RISC-V RV32IMAFC instruction set
+- 4KB instruction cache size
+- 256KB SDRAM in total
```
Comment (tari) [inline boards/riscv/it82xx2_evb/doc/index.rst:28]: It's SRAM, not SDRAM, right?

File: drivers/interrupt_controller/intc_ite_it8xxx2_v2.c [hunk 22]
```diff
@@ -0,0 +1,245 @@
...
+
+	/* Critical section due to run a bit-wise NAND operation */
+	unsigned int key = irq_lock();
+
+	IT8XXX2_INTC_IER(group) &= ~BIT(index);
+	/*
+	 * This load operation will guarantee the above modification of
+	 * SOC's register can be seen by any following instructions.
+	 */
+	_ier = IT8XXX2_INTC_IER(group);
```
Comment (tari) [inline drivers/interrupt_controller/intc_ite_it8xxx2_v2.c:133]: You should be able to remove `_ier` and keep only the read, since that's a volatile read. `_ier` forces the unnecessary stack allocation of a variable that gets written but is otherwise unused.

    /*
     * This load operation will guarantee the above modification of
	 * SOC's register can be seen by any following instructions.
	 */
    IT8XXX2_INTC_IER(group);

(Same above in `ite_intc_save_and_disable_interrupts`.)

File: drivers/pinctrl/pinctrl_ite_it8xxx2.c [hunk 23]
```diff
@@ -308,6 +326,16 @@ static int pinctrl_it8xxx2_init(const struct device *dev)
 	 * have to set UART1PSEL = 1 in UART1PMR register.
 	 */
 
+#ifdef CONFIG_SOC_IT8XXX2_REG_SET_V2
+	/*
+	 * Swap the default I2C2 SMCLK2/SMDAT2 pins from GPC7/GPD0 to GPF6/GPF7,
```
Comment (tari) [inline drivers/pinctrl/pinctrl_ite_it8xxx2.c:331]: Should this be optional? It seems odd to unconditionally remap some of the IOs.

File: boards/riscv/it82xx2_evb/doc/index.rst (hunk 21 above)
Comment (GTLin08) [reply boards/riscv/it82xx2_evb/doc/index.rst]: Yes, fixed.

File: drivers/interrupt_controller/intc_ite_it8xxx2_v2.c (hunk 22 above)
Comment (GTLin08) [reply drivers/interrupt_controller/intc_ite_it8xxx2_v2.c]: Removed.

**Comment by GTLin08** [comment]
Rebased. Resolve conflicting in soc/riscv/riscv-ite/it8xxx2/soc.c

File: soc/riscv/riscv-ite/it8xxx2/ilm.c
Comment (keith-zephyr) [inline soc/riscv/riscv-ite/it8xxx2/ilm.c:1]: This change removes the initialization of the RVILMCR0 register.  Isn't setup of RVILMCR0 still required?

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: dts/riscv/ite/it82xx2.dtsi [hunk 24]
```diff
@@ -24,6 +24,11 @@
 			interrupt-parent = <&intc>;
 		};
 
+		gpiogcr: gpio-gcr@f03e00 {
+			compatible = "ite,it8xxx2-gpiogcr";
+			reg = <0x00f03e00 0x100>;
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply dts/riscv/ite/it82xx2.dtsi]: Done.

File: drivers/gpio/gpio_ite_it8xxx2_v2.c [hunk 25]
```diff
@@ -0,0 +1,460 @@
...
+
+static int gpio_ite_pin_interrupt_configure(const struct device *dev,
+					    gpio_pin_t pin,
+					    enum gpio_int_mode mode,
+					    enum gpio_int_trig trig)
+{
+	const struct gpio_ite_cfg *gpio_config = dev->config;
+	uint8_t gpio_irq = gpio_config->gpio_irq[pin];
+
+	if (mode == GPIO_INT_MODE_DISABLED) {
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply drivers/gpio/gpio_ite_it8xxx2_v2.c:342]: Done.

File: drivers/pinctrl/pinctrl_ite_it8xxx2.c (hunk 23 above)
Comment (GTLin08) [reply drivers/pinctrl/pinctrl_ite_it8xxx2.c:331]: Yes, that is correct. 

I have created these mapped pins to the it8xxx2-pinctrl-map.dtsi file.
Added i2c2_clk_gpc7_default/i2c2_data_gpd0_default, 
            i2c3_clk_gpb2_default/i2c3_data_gpb5_default
            i2c5_clk_gpe1_default/i2c5_data_gpe2_default

If the user selects GPC7/GPD0, the as configuration below will select IT8XXX2_ALT_FUNC_4.
And the `pinctrl_gpio_it8xxx2_configure_pins` function in pinctrl_ite_it8xxx2.c will make the corresponding set.
The `IT8XXX2_GPIO_SMB2PS `bit will get set.
```
	i2c2_clk_gpc7_default: i2c2_clk_gpc7_default {
		pinmuxs = <&pinctrlc 7 IT8XXX2_ALT_FUNC_4>;
	};
	i2c2_data_gpd0_default: i2c2_data_gpd0_default {
		pinmuxs = <&pinctrld 0 IT8XXX2_ALT_FUNC_4>;
	};
```

File: soc/riscv/riscv-ite/it8xxx2/ilm.c
Comment (GTLin08) [reply soc/riscv/riscv-ite/it8xxx2/ilm.c:1]: It has been confirmed that RVILMCR0 setup is not required.

**Comment by GTLin08** [comment]
Rebased. Resolve conflicting in test/drivers/adc/adc_api/src/test_adc.c

`config ILM_MAX_SIZE` moves from Kconfig.defconfig.series to Kconfig.soc will pass Compliance Checks

File: drivers/gpio/gpio_ite_it8xxx2_v2.c [hunk 26]
```diff
@@ -0,0 +1,460 @@
...
+	.reg_gpdr = DT_INST_REG_ADDR_BY_IDX(inst, 0),              \
+	.reg_gpdmr = DT_INST_REG_ADDR_BY_IDX(inst, 1),             \
+	.reg_gpotr = DT_INST_REG_ADDR_BY_IDX(inst, 2),             \
+	.reg_p18scr = DT_INST_REG_ADDR_BY_IDX(inst, 3),            \
+	.reg_gpcr = DT_INST_REG_ADDR_BY_IDX(inst, 4),              \
+	.wuc_base = DT_INST_PROP_OR(inst, wuc_base, {0}),          \
+	.wuc_mask = DT_INST_PROP_OR(inst, wuc_mask, {0}),          \
+	.gpio_irq = IT8XXX2_DT_GPIO_IRQ_LIST(inst),                \
+	.has_volt_sel = DT_INST_PROP_OR(inst, has_volt_sel, {0}),  \
+	.index = inst,                                             \
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply drivers/gpio/gpio_ite_it8xxx2_v2.c]: Remove `.index` and add `.dev_name`.

File: soc/riscv/riscv-ite/it8xxx2/ilm.c [hunk 27]
```diff
@@ -50,6 +50,12 @@ BUILD_ASSERT((ILM_BLOCK_SIZE & (ILM_BLOCK_SIZE - 1)) == 0, "ILM_BLOCK_SIZE must
 #define SCARH_ENABLE	 BIT(3)
 #define SCARH_ADDR_BIT19 BIT(7)
 
+#if defined(CONFIG_SOC_IT81202_CX) || defined(CONFIG_SOC_IT81302_CX)
+#define ILM_MAX_SIZE 60
+#else
+#define ILM_MAX_SIZE RAM_SIZE
+#endif
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply soc/riscv/riscv-ite/it8xxx2/ilm.c]: Thank you for your suggestion.
`$(dt_chosen_reg_size_int,$(DT_CHOSEN_Z_SRAM),0,K)` can be replaced as follows:
```
config ILM_MAX_SIZE
	int "ILM Size in kB"
	default 60 if SOC_IT81202_CX || SOC_IT81302_CX
	default SRAM_SIZE
```

File: drivers/pinctrl/pinctrl_ite_it8xxx2.c (hunk 23 above)
Comment (keith-zephyr) [reply drivers/pinctrl/pinctrl_ite_it8xxx2.c:331]: Thank you for the explanation.

**Review by keith-zephyr — COMMENTED** [review]
One last comment from me.  Eliminate the dev_name from the GPIO driver.

File: drivers/gpio/gpio_ite_it8xxx2_v2.c [hunk 28]
```diff
@@ -0,0 +1,459 @@
...
+		*reg_gpcr = GPCR_PORT_PIN_MODE_TRISTATE;
+		/*
+		 * Since not all GPIOs can be to configured as tri-state,
+		 * prompt error if pin doesn't support the flag.
+		 */
+		if (*reg_gpcr != GPCR_PORT_PIN_MODE_TRISTATE) {
+			/* Go back to default setting (input) */
+			*reg_gpcr = GPCR_PORT_PIN_MODE_INPUT;
+			LOG_ERR("Cannot config the node-%s, pin=%d as tri-state",
+				gpio_config->dev_name, pin);
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply drivers/gpio/gpio_ite_it8xxx2_v2.c]: Done.
The `reg_gpdr` variable has been defined above, which can be used in place of the `gpio_config->reg_gpdr`.

**Review by keith-zephyr — APPROVED** [review]

**Review by Dino-Li — APPROVED** [review]
